EventListener/UserContextSubscriber.php

```diff
@@ -124,6 +126,15 @@ public function onKernelResponse(FilterResponseEvent $event)
         }
 
         $response = $event->getResponse();
+        $request = $event->getRequest();
+
+        // hash has changed, session has most certainly changed, prevent setting incorrect cache
+        if ($request->attributes->get($this->hashHeader) !== $request->headers->get($this->hashHeader)) {
```

Comment:
below we have a different logic: if the hashHeader is not present, we vary on the userIdentifierHeaders. i am afraid this line could break at least the hash lookup request.

looking at the code below, i wonder why we force varying on the hash. there could be parts of the site that do not depend on the user and don't want to vary on the hash... but that question is out of scope for this PR.

maybe only do the sanity check if there was a hash header in the request, in the `if` case below?